### PR TITLE
Initial merge of examples repo into library

### PR DESCRIPTION
**Pull request by byrnHDF**

### CODE REVIEW

File: HDF5Examples/COPYING
Comment (jhendersonHDF) [inline HDF5Examples/COPYING:1]: Probably not needed since we have the root one?

File: HDF5Examples/JAVA/intro/testfiles/116/HDF5DatasetCreate.txt
Comment (jhendersonHDF) [inline HDF5Examples/JAVA/intro/testfiles/116/HDF5DatasetCreate.txt:1]: There seem to be several empty files like this under this folder

**Comment by brtnfld** [comment]
Did we change that much between versions so we can't use the same test reference files? Can't we ignore inconsequential differences?

File: HDF5Examples/COPYING
Comment (byrnHDF) [reply HDF5Examples/COPYING:1]: No, we need it for the binaries - I use the HDF5Example folder as a standalone build with the binaries.

File: HDF5Examples/JAVA/intro/testfiles/116/HDF5DatasetCreate.txt
Comment (byrnHDF) [reply HDF5Examples/JAVA/intro/testfiles/116/HDF5DatasetCreate.txt:1]: By-product of being able to use HDF5Examples as a general solution for hdf5 versions (including develop) and API restrictions.

**Comment by byrnHDF** [comment]
> Did we change that much between versions so we can't use the same test reference files? Can't we ignore inconsequential differences?

Not without a lot of development time.

**Pull request closed without merging**